pkg/operator/controller/ingress/controller.go

```diff
@@ -559,6 +563,35 @@ func filterTLS13Ciphers(ciphers []string) []string {
 	return filteredCiphers
 }
 
+// validateHTTPHeaderBuffer validates the given ingresscontroller's header buffer
+// size configuration, if it specifies one.
+func validateHTTPHeaderBufferValues(ic *operatorv1.IngressController) error {
+	if (ic.Spec.HTTPHeaderBuffer == operatorv1.IngressControllerHTTPHeaderBuffer{}) {
+		return nil
+	}
+
+	bufSize := int(ic.Spec.HTTPHeaderBuffer.HeaderBufferBytes)
+	maxRewrite := int(ic.Spec.HTTPHeaderBuffer.HeaderBufferMaxRewriteBytes)
+
+	// HeaderBufferBytes and HeaderBufferMaxRewriteBytes are both
+	// optional fields. Substitute the default values used by the
+	// router when either field is empty so that we can ensure that
+	// tune.maxrewrite will never wind up being greater than tune.bufsize
+	// (which would break router reloads).
+	if bufSize == 0 {
+		bufSize = routerDefaultHeaderBufferSize
+	} else if maxRewrite == 0 {
```

Comment:
Should this be a discrete check (i.e., drop the `else` here)?

Reply:
No reason not to I suppose. Fixed.